src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp

```diff
@@ -137,6 +137,32 @@ void mark_runtime_skippable_nodes::run(program& p) {
                     }
                 }
 
+                node.can_be_optimized(true);
+                // Set runtime skippable only when the node is set as can_be_optimized finally.
+                node.set_runtime_skippable(true);
+                GPU_DEBUG_TRACE_DETAIL << "[mark_runtime_skippable_nodes] : " << node.id() << " can_be_optimized" << std::endl;
+            }
+        });
+        program_helpers::do_for_types<reorder>(*node, [](reorder_node& node){
+            auto impl_params = node.get_kernel_impl_params();
+
+            if ((node.is_output() && node.get_dependency(0).is_input())
+                || node.has_fused_primitives()
+                || (impl_params->get_input_layout(0).format != impl_params->get_output_layout().format)
+                || (impl_params->get_input_layout(0).data_type != impl_params->get_output_layout().data_type)
+                || (impl_params->get_input_layout(0).data_padding != impl_params->get_output_layout().data_padding))
+                return;
+
+            // TODO: For now, all reorders with dynamic shape are applied.
+            //       A more detailed pattern will need to be applied later
+            if (node.is_dynamic()) {
+                if (!node.is_output() && node.get_users().size() != 1)
+                    return;
+
+                // If the user is concatenation with 1 user, priority should be given to in place concat optimization at runtime
+                if (node.have_user_with_type<concatenation>())
```

Comment:
optimizable concat user + and the concat user is the only user

Reply:
Done